### ISSUE TITLE
Clean up scroll behavior in data choices list plus make dataChoices behavior globally accessible

### ISSUE THREAD

**Issue by NealHumphrey** [codefordc/housing-insights#468]
In the data choices list on the sidebar, selecting a new data choice collapses the open data control and opens the selected control (due to magic of semantic ui's accordion). However, if the data choice is not near the top of the list, it can appear open below the current scroll level, causing the user to have to manually scroll down. 

- [ ] in the clickCallback on row 775, use setState to announce the selected data source i.e. `setState('dataChoice',c.source)`. This will also give the URL encoding something to tap into for saving which dataChoice was open in the view. 

- [ ] Refactor the clickCallback into a globally accessible function `selectDataChoice` so that it can be called from elsewhere (this sets us up for url decoding, since we're working on this anyways). 
- [x] Auto-scroll the sidebar so that the selected data choice is at the top of the visible pane when that function is run. 
- [x] Add extra padding of 150 pixels below the `#filter-components` div. This combined with autoscroll will solve the scroll item in #396

**Comment by NealHumphrey** [comment]
You can test the issue in #396 by moving the 'portfolio' choice to the bottom of the list in `dataChoices.js` (they are rendered in order). the dropdown appears out of view.

**Comment by ostermanj** [comment]
I wonder if we should allow more than one accordion to be open at a time. I could see that be more expected and certainly gives more control to the users, but if no one's mentioned it perhaps it's not an issue. It would be nice when decoding filter settings from a url to show the active filters, for instance. It wold affect the behavior of showing  data overlays, which would have to activate only on clicks that open the associated according. Currently activates on click. Thoughts?

**Comment by ostermanj** [comment]
I've addressed this issue in later commits of PR #478 . I included it there to avoid conflicting with it. The mechanisms are a little different than laid out in the checklist above, with the main difference being that only overLay dataChoices are being url encoded and decoded. That functionality could be extended to non-overLay dataChoices, but unless their filters are activated there doesn't seem to be reason to do so, and any activated filters are already being encoded.

Given that, I'm using the existing `setState(overLay, ...)` to trigger the state encoding rather than creating a new `setState(dataChoice, ...)`.

The auto scroll is handled via an onOpen callback method native to the semantic ui accordions rather than in the clickCallback.

The overLay decoding programmatically 'clicks' the associated accordion element.

**Comment by NealHumphrey** [comment]
@ostermanj Great! sounds good on using the `overlay` state instead of making a new one - you're right that we don't really need to use the open-accordion functionality for project-based data sources, just overlay ones. 

RE: your earlier question on more than one accordion open at a time. The reason I switched it to just one was b/c of the overlay issue - in v1 when we had both overlays and filters, overlays had one open at a time (and revealed the legend) and filters could have multiple ones. I think since no one has asked for it, we should keep as is since allowing more than one open means we need to come up with some sort of different behavior for how overlays will be toggled. 

In v1, I also had activated filters change their dropdown header band color to be dark orange if they were activated, while they are the (current) light orange if they are not in use. Users in round 1 of testing didn't really notice that so I removed it, but I think that with the new behavior of the dark orange pillboxes rising from their source location it might now be clearer to users - and as you mention, a good visual cue when loading pre-filtered data from a url since hte user won't have past behavior to tell them what had been selected. I think that I will
a) restore the dark-orange highlighting fora ctivated filters and 
b) make the pillboxes clickable so that if you click the pillbox (not the x) it will open that filter

Hopefully those two changes will make understanding what is contained in a loaded url filter set more understandable.
